src/XrdCl/XrdClFileStateHandler.cc

```diff
@@ -1858,6 +1858,52 @@ namespace XrdCl
     return SendOrQueue( self, *self->pDataServer, msg, stHandler, params );
   }
 
+
+  //----------------------------------------------------------------------------
+  // Performs a custom operation on an open file, server implementation
+  // dependent - async
+  //----------------------------------------------------------------------------
+  XRootDStatus FileStateHandler::Fcntl( std::shared_ptr<FileStateHandler> &self,
```

Comment:
I don't understand this change. Technically, it should have been a change to two lines in the existing FileStateHandler::Fcntl()
1) The method definitions and
2) Replacing line XrdClFileStateHandler.cc:1887 to use the query code passed via the argument list.
That's a minor change and this looks way too much of a change than was needed.